### ISSUE TITLE
Unexpected Memory Problem

### ISSUE THREAD

**Issue by ffllyy467** [HeegerGao/VLA-OS#8]
Hi,

Thanks for your excellent work. Currently, I'm trying to follow your instructions to train VLA-OS in my own scene, but when I'm fine-tuning the integrated VLA I met an unexpected memory problem on my server, which has 8x H100 cards, 1600 GB RAM, and 120 CPU Cores. During training, the memory usage continuously expands to the limit, and the training process is killed, as shown in the picture:

<img width="1233" height="336" alt="Image" src="https://github.com/user-attachments/assets/30f6c7f3-da07-48a7-a007-4f25ab3dfc99" />

I'm using my own dataset, with 50 trajectories and ~250 steps per trajectory, with 2 images of 640x480 in the raw data, and I strictly follow the instructions in [this repo](https://github.com/Panda-Shawn/VLA-OS-Language-Planning-Labeling) to process my data. Here is my training configuration that may help with this problem:
```yaml
data:
  root_dir: <my_dataset_dir>
  data_mix: <my_data_mix>
  shuffle_buffer_size: 8_000
  image_aug: true
  future_action_window_size: 7
  goal_image_step: 64  # goal image step
  image_window_size: 2
  load_depth: false
  use_wrist_image: true
  use_proprio: true
  given_camera_views: null

training:
  stage: full-finetune
  epochs: 500
  max_steps: 150_000
  global_batch_size: 16
  per_device_batch_size: 2
  learning_rate: 1e-4
  weight_decay: 0.0
  max_grad_norm: 1.0
  lr_scheduler_type: constant
  warmup_ratio: 0.03
  sharding_strategy: full-shard
  save_step: 1000
  training_algo: bc
  # training_algo: flow_matching
  skewed_timesteps: false
  planning_loss_weight: 1.0
  action_loss_weight: 1.0
```
where I decrease the batch size and shuffle_buffer_size to avoid memory and CUDA memory exhaustion during setup. However, my training process gradually ate up all 1600GB of memory on the server and was killed after ~4000 steps before it exhausted all the memory, and I couldn't find out how it was leaked. Could you please share some key configurations and techniques you use, or the code that may be relevant to this memory problem? Thanks a lot for your help!

**Comment by HeegerGao** [comment]
Thanks for your question. I will come to this when I have time.
